scene/gui/control.cpp

```diff
@@ -616,23 +611,13 @@ Window *Control::get_parent_window() const {
 
 Control *Control::get_root_parent_control() const {
 	ERR_READ_THREAD_GUARD_V(nullptr);
-	const CanvasItem *ci = this;
-	const Control *root = this;
-
-	while (ci) {
-		const Control *c = Object::cast_to<Control>(ci);
-		if (c) {
-			root = c;
-
-			if (c->data.RI || c->is_top_level_control()) {
-				break;
-			}
+	const Control *c = this;
+	if (is_inside_tree()) {
+		while (c->data.parent_control) {
+			c = c->data.parent_control;
```

Comment:
`data.parent_control` doesn't consider being `top_level` (it's set using `data.parent_control = Object::cast_to<Control>(parent_node)`), so as far as I can tell this changes behavior.